dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java

```diff
@@ -197,25 +196,22 @@ private ResultSet executeQuery(String query) throws Exception {
     public void write(String query) throws Exception {
       checkArgument(dataFileWriter != null,
                     "Avro DataFileWriter was not properly created");
+      RowMapper avroRowMapper = jdbcAvroOptions.getAvroRowMapper();
+      Schema schema = coder.getSchema();
       logger.info("jdbcavroio : Starting write...");
       try (ResultSet resultSet = executeQuery(query)) {
         checkArgument(resultSet != null,
                       "JDBC resultSet was not properly created");
         this.writeIterateStartTime = System.currentTimeMillis();
         while (resultSet.next()) {
-          GenericRecord genericRecord =
-              jdbcAvroOptions.getAvroRowMapper().convert(resultSet, coder.getSchema());
-          writeRecord(genericRecord);
+          GenericRecord genericRecord = avroRowMapper.convert(resultSet, schema);
+          this.dataFileWriter.append(genericRecord);
```

Comment:
The four changes happening here seem to be refactoring and not effecting the logical behavior, and could be separated into different commit.

Reply:
It is in a separate commit. Do you mean a separate PR also?

Reply:
No, you are right. I was reviewing in hurry. Sorry, all hood here.